src/ExampleApp.cpp

```diff
@@ -117,143 +132,316 @@ ExampleApp::ExampleApp(Eegeo::EegeoWorld* pWorld,
-	m_pExampleController->RegisterCameraExample<Examples::SingleCityExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::ResourceSpatialQueryExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::RouteDrawingExampleFactory>();
+    m_pExampleController->RegisterCameraControllerScreenPropertiesProviderExample<Examples::RouteSimulationAnimationExampleFactory>(m_screenPropertiesProvider);
+    m_pExampleController->RegisterCameraExample<Examples::RouteThicknessPolicyExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::ScreenPickExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::ScreenUnprojectExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::SingleCityExampleFactory>();
     m_pExampleController->RegisterCameraExample<Examples::StencilAreaExampleFactory>();
-	m_pExampleController->RegisterCameraExample<Examples::ToggleTrafficExampleFactory>();
-	m_pExampleController->RegisterCameraExample<Examples::TrafficCongestionExampleFactory>();
-	m_pExampleController->RegisterCameraExample<Examples::WebRequestExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::ToggleTrafficExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::TrafficCongestionExampleFactory>();
+    m_pExampleController->RegisterCameraExample<Examples::WebRequestExampleFactory>();
     m_pExampleController->RegisterCameraControllerScreenPropertiesProviderExample<Examples::RenderToTextureExampleFactory>(m_screenPropertiesProvider);
+	#ifdef CARDBOARD
+    m_pExampleController->RegisterScreenPropertiesProviderVRExample<Examples::VRCardboardExampleFactory>(m_screenPropertiesProvider);
+	#endif
 }
 
 ExampleApp::~ExampleApp()
 {
-	delete m_pCameraTouchController;
+    delete m_pCameraTouchController;
     delete m_pLoadingScreen;
     delete m_pExampleController;
+	#ifdef CARDBOARD
+    Eegeo_DELETE m_pVRSkybox;
+    Eegeo_DELETE m_pVRDistortion;
+
+    Eegeo_DELETE m_pStreamingVolume;
+	#endif
 }
 
 void ExampleApp::OnPause()
 {
-	Eegeo::EegeoWorld& eegeoWorld = World();
-	eegeoWorld.OnPause();
+    Eegeo::EegeoWorld& eegeoWorld = World();
+    eegeoWorld.OnPause();
 }
 
 void ExampleApp::OnResume()
 {
-	Eegeo::EegeoWorld& eegeoWorld = World();
-	eegeoWorld.OnResume();
+    Eegeo::EegeoWorld& eegeoWorld = World();
+    eegeoWorld.OnResume();
 }
 
-void ExampleApp::Update (float dt)
+void ExampleApp::Update (float dt, float headTansform[])
 {
-	Eegeo::EegeoWorld& eegeoWorld = World();
-    
+    Eegeo::EegeoWorld& eegeoWorld(World());
+    eegeoWorld.EarlyUpdate(dt);
+    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
+
     m_pCameraTouchController->Update(dt);
 
-	eegeoWorld.EarlyUpdate(dt);
-    
-	m_pExampleController->EarlyUpdate(dt);
-    
-    Eegeo::Camera::CameraState cameraState(m_pExampleController->GetCurrentCameraState());
```

Comment:
Beyond this point I'd expect the information in headTransform[] to have been encapsulated in the camera state. From there I wouldn't expect us to need the headTransform[] again in this frame.